src/components/interactive-builder/modals/new-section/section.modal.tsx

```diff
@@ -10,37 +10,64 @@ interface SectionModalProps {
   schema: Schema;
   onSchemaChange: (schema: Schema) => void;
   pageIndex: number;
+  sectionIndex?: number;
+  modalType?: 'edit';
 }
 
-const SectionModal: React.FC<SectionModalProps> = ({ closeModal, schema, onSchemaChange, pageIndex }) => {
+const SectionModal: React.FC<SectionModalProps> = ({
+  closeModal,
+  schema,
+  onSchemaChange,
+  pageIndex,
+  sectionIndex,
+  modalType,
+}) => {
   const { t } = useTranslation();
-  const [sectionTitle, setSectionTitle] = useState('');
+  const [sectionTitle, setSectionTitle] = useState(() => {
+    return modalType === 'edit' ? schema.pages[pageIndex].sections[sectionIndex].label : '';
+  });
```

Comment:
This isn't a heavy computation direct initialization is better
```suggestion
const [sectionTitle, setSectionTitle] = useState<string>(
  modalType === 'edit' ? schema.pages[pageIndex].sections[sectionIndex].label : ''
);
```